feat: API to replace email alias generation commands

jennifer-richards opened the pull request:
This provides the API endpoints and script needed to remove email alias management commands from Cron-runner

jennifer-richards committed:
feat: DraftAliasGenerator class

Encapsulates logic from generate_draft_aliases.py

jennifer-richards committed:
refactor: Avoid circular imports

jennifer-richards committed:
feat: Add draft_aliases API endpoint

jennifer-richards committed:
feat: Add @requires_api_token decorator

Stolen from feat/rpc-api

jennifer-richards committed:
feat: Add token auth to draft_aliases endpoint

jennifer-richards committed:
feat: draft-aliases-from-json.py script

Parses output from the draft_aliases API call

jennifer-richards committed:
chore: Remove unused cruft

jennifer-richards committed:
refactor: Avoid shadowing "draft" name

jennifer-richards committed:
fix: Suppress empty lists from DraftAliasGenerator

jennifer-richards committed:
refactor: Use a GET instead of POST

jennifer-richards committed:
feat: GroupAliasGenerator class

jennifer-richards committed:
feat: group aliases API view

jennifer-richards committed:
fix: Handle domains array correctly

jennifer-richards committed:
fix: Suppress empty group aliases

jennifer-richards committed:
refactor: Generalize aliases-from-json.py script

jennifer-richards committed:
refactor: Same output fmt for draft and group alias apis

jennifer-richards committed:
feat: Sort addresses for stability

jennifer-richards committed:
fix: Add "anything" virtual alias

IvanVnn commented:
> This provides the API endpoints and script needed to remove email alias management commands from Cron-runner

jennifer-richards committed:
Merge branch 'main' into celery-cronch-runner

jennifer-richards committed:
test: Test requires_api_token decorator

jennifer-richards committed:
feat: Harden is_valid_token against misconfig

jennifer-richards committed:
test: Test is_valid_token

jennifer-richards committed:
test: Test draft_aliases view

jennifer-richards committed:
test: Test group_aliases view

jennifer-richards committed:
test: Test DraftAliasGenerator

jennifer-richards committed:
fix: ise group is type "ise" in test data

jennifer-richards committed:
test: Fix logic in testManagementCommand

The test was incorrect - and fails when fixed. :-(

jennifer-richards committed:
test: Test GroupAliasGenerator

Test currently fails

jennifer-richards committed:
fix: Suppress empty -ads alias

jennifer-richards committed:
test: Fix group acronym copy/paste error

I *think* this must be what had been intended. The
code does not look like it ever dealt with GroupHistory,
so I'm pretty sure it wasn't meant to have the same
acronym used by two different Groups at different
times.

jennifer-richards committed:
test: Check draft .notify alias generation

jennifer-richards committed:
test: Cover get_draft_notify_emails()

jennifer-richards committed:
Merge branch 'main' into celery-cronch-runner
